test/blackbox/api/dds-pim/PubSubReader.hpp

```diff
@@ -471,36 +515,28 @@ class PubSubReader
     {
         if (participant_ != nullptr)
         {
-            if (datareader_)
-            {
-                subscriber_->delete_datareader(datareader_);
-                datareader_ = nullptr;
-            }
-            if (subscriber_)
-            {
-                participant_->delete_subscriber(subscriber_);
-                subscriber_ = nullptr;
-            }
-            if (topic_)
-            {
-                participant_->delete_topic(topic_);
-                topic_ = nullptr;
-            }
-            eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant_);
+            ASSERT_EQ(ReturnCode_t::RETCODE_OK, participant_->delete_contained_entities());
+            datareader_ = nullptr;
+            subscriber_ = nullptr;
+            cf_topic_ = nullptr;
+            topic_ = nullptr;
+
+            ASSERT_EQ(ReturnCode_t::RETCODE_OK,
+                    eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(participant_));
             participant_ = nullptr;
         }
 
         initialized_ = false;
     }
 
-    std::list<type> data_not_received()
+    std::list<typename TypeTraits::DataListType> data_not_received()
```

Comment:
Would improve the readability if we define an alias `using typesupport_type = typename TypeTraits::DataListType` and we use `typesupport_type` within the rest of the code ? What do you think ?

This would apply for all the `PubSubHeaders`